server/src/main/java/org/elasticsearch/common/logging/DeprecatedMessage.java

```diff
@@ -40,10 +41,14 @@ public static ESLogMessage of(String key, String xOpaqueId, String messagePatter
             @Override
             public String toString() {
                 return ParameterizedMessage.format(messagePattern, args);
-
             }
         };
+
         return new ESLogMessage(messagePattern, args)
+            .field("data_stream.type", "logs")
+            .field("data_stream.datatype", "deprecation")
```

Comment:
@pugnascotia This should be `data_stream.dataset` to be aligned with the indexing strategy.

I would also propose to keep the namespace as `default` and use `deprecation.elasticsearch` as the dataset name. Only important thing is that the dataset does not contain a  `-`.

Reply:
@ruflin so would we have the following, then?

```
    .field("data_stream.dataset", "default")
    .field("data_stream.namespace", "deprecation.elasticsearch")
```

Reply:
No, the other way around:
```
.field("data_stream.dataset", "deprecation.elasticsearch")
.field("data_stream.namespace", "default")
```

Reply:
Thanks, I'll get that changed 👍